core/src/main/java/overflowdb/Graph.java

```diff
@@ -0,0 +1,296 @@
+    }
+    if (nodes.contains(id)) {
+      throw new IllegalArgumentException(String.format("Vertex with id already exists: %s", id));
+    }
+
+    currentId.set(Long.max(id, currentId.get()));
+    final NodeRef node = createNode(id, label, keyValues);
+    nodes.add(node);
+    return node;
+  }
+
+  private NodeRef createNode(final long idValue, final String label, final Object... keyValues) {
+    if (!nodeFactoryByLabel.containsKey(label)) {
+      throw new IllegalArgumentException("No NodeFactory for label=" + label + " available.");
+    }
+    final NodeFactory factory = nodeFactoryByLabel.get(label);
+    final NodeDb node = factory.createNode(this, idValue);
+    PropertyHelper.attachProperties(node, keyValues);
+    this.referenceManager.registerRef(node.ref);
+
+    return node.ref;
+  }
+
+  @Override
+  public String toString() {
+    return String.format("%s [%d nodes]", getClass().getSimpleName(), nodeCount());
+  }
+
+  /**
+   * if the config.graphLocation is set, data in the graph is persisted to that location.
+   */
+  @Override
+  public void close() {
+    this.closed = true;
+    heapUsageMonitor.ifPresent(monitor -> monitor.close());
+    if (config.getStorageLocation().isPresent()) {
+      /* persist to disk */
+      indexManager.storeIndexes(storage);
+      referenceManager.clearAllReferences();
+    }
+    referenceManager.close();
+    storage.close();
+  }
+
+  public int nodeCount() {
+    return nodes.size();
+  }
+
+  public int nodeCount(String label) {
+    return nodes.cardinality(label);
+  }
+
+  public int edgeCount() {
+    int i = 0;
+    final Iterator<Edge> edges = edges();
+    while (edges.hasNext()) {
+      edges.next();
+      i++;
+    }
+    return i;
+  }
+
+  /** Iterator over all edges - alias for `edges` */
+  public Iterator<Edge> E() {
+    return edges();
+  }
+
+  /** Iterator over all edges */
+  public Iterator<Edge> edges() {
+    return IteratorUtils.flatMap(nodes(), node -> node.outE());
+  }
+
+  /** Iterator over edges with given label */
+  public Iterator<Edge> edges(String label) {
+    return IteratorUtils.flatMap(nodes(), node -> node.outE(label));
+  }
+
+  /** Iterator over all nodes - alias for `nodes` */
+  public Iterator<Node> V() {
+    return nodes();
+  }
+
+  /** Iterator over all nodes */
+  public final Iterator<Node> nodes() {
+    return nodes.iterator();
+  }
+
+  /** Iterator over nodes with provided ids - alias for `nodes(ids...)`
+   * note: does not return any nodes if no ids are provided */
+  public Iterator<Node> V(long... ids) {
+    return nodes(ids);
+  }
+
+  public final Node node(long id) {
+    return nodes.nodeById(id);
+  }
+
+  /** Iterator over nodes with provided ids
+   * note: this behaves differently from the tinkerpop api, in that it returns no nodes if no ids are provided */
+  public final Iterator<Node> nodes(long... ids) {
+    if (ids.length == 0) {
```

Comment:
it wouldn't hurt to do so, but how would the method call for `nodes` look like to trigger this? i tried from java and scala

Reply:
i think i get it now - the reason it's not possible to pass `null` is because `nodes` is overloaded. 
so it's kinda flakey, but the same as below applies: if you pass null, you deserve a NPE :)